### ISSUE TITLE
Add library drag and drop sorting

### ISSUE THREAD

**Issue by AbdelrahmanElnezamy121** [mihonapp/mihon#210]
### Describe your suggested feature

Add library drag and drop sorting

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have updated the app to version **[0.16.1](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by MajorTanya** [comment]
Care to elaborate?

**Comment by AbdelrahmanElnezamy121** [comment]
> > Care to elaborate? 
> 
> They probably mean a custom sort order where you can freedly drag and drop your entries from one position to another and it stays in that position. 

It's what I mean

**Comment by joaoandre120** [comment]
Before, in the settings it was possible to do this, it was much more practical.

**Comment by MartsTech** [comment]
I believe it should be possible to use some implementation of https://github.com/aclassen/ComposeReorderable for the drag/drop. Then in the library tab when the user is in selection mode, an option in the topbar for custom order could be added. This will allow the user to drag/drop instead of select more items and the order idx will be stored in the library item model. Finally a custom sort option could be added in the settings dialog.

**Comment by MajorTanya** [comment]
> Before, in the settings it was possible to do this, it was much more practical.

Yes, this was omitted when the screen in question was migrated to the new Compose UI framework. My personal guess is that it would have been a nightmare to properly implement.

> I believe it should be possible to use some implementation of https://github.com/aclassen/ComposeReorderable for the drag/drop. Then in the library tab when the user is in selection mode, an option in the topbar for custom order could be added. This will allow the user to drag/drop instead of select more items and the order idx will be stored in the library item model. Finally a custom sort option could be added in the settings dialog.

That looks promising but as with every feature for Mihon, someone will have to take this up and PR it in order for it to happen 👍 Thanks for the research and recommendation
